src/placeos-core/module_manager.cr

```diff
@@ -275,37 +275,54 @@ module PlaceOS::Core
     # Clustering
     ###############################################################################################
 
-    def on_managed_edge?(mod : Model::Module)
-      mod.on_edge? && own_node?(mod.edge_id.as(String))
-    end
-
-    private getter semaphore = Atomic(Int32).new(0)
+    private getter queued_stabilization_events = Atomic(Int32).new(0)
     private getter stabilize_lock = Mutex.new
+    private STABILIZE_BATCH_SIZE = 32
 
     # Run through modules and load to a stable state.
     #
     # Uses a semaphore to ensure intermediary cluster events don't trigger stabilization.
     def stabilize(nodes : Array(HoundDog::Service::Node)) : Bool
-      semaphore.add(1)
+      queued_stabilization_events.add(1)
       stabilize_lock.synchronize do
-        semaphore.add(-1)
-        return false unless semaphore.get.zero?
+        queued_stabilization_events.add(-1)
+        return false unless queued_stabilization_events.get.zero?
 
         Log.debug { {message: "stabilizing", nodes: nodes.to_json} }
 
         # Create a one off rendezvous hash with nodes from the stabilization event
         rendezvous_hash = RendezvousHash.new(nodes: nodes.map(&->HoundDog::Discovery.to_hash_value(HoundDog::Service::Node)))
-        Model::Module.all.each do |m|
-          begin
-            load_module(m, rendezvous_hash)
-          rescue e
-            Log.error(exception: e) { {message: "failed to load module during stabilization", module_id: m.id, name: m.name, custom_name: m.custom_name} }
+
+        success_count, fail_count = 0_i64, 0_i64
```

Comment:
```suggestion
        success_count, fail_count = 0_u32, 0_u32
```

Reply:
`u32` is not an accepted `Log::Metadata::Value`, [see here](https://github.com/crystal-lang/crystal/blob/a4a371bf74598e6cc025d18d8a309a46328718c0/src/log/metadata.cr#L199)
While it would be more correct type wise, this saves an intermediate conversion to string or alias into another int type